great_expectations/agent/agent.py

```diff
@@ -212,29 +218,26 @@ def _reject_correlation_id(self, id: str):
             self._correlation_ids.clear()
         return should_reject
 
-    @classmethod
-    def _get_config(cls) -> GXAgentConfig:
+    def _get_config(self) -> GXAgentConfig:
         """Construct GXAgentConfig."""
 
         # ensure we have all required env variables, and provide a useful error if not
-
-        class GxAgentConfigSettings(pydantic.BaseSettings):
-            gx_cloud_base_url: str = "https://api.greatexpectations.io"
-            gx_cloud_organization_id: str
-            gx_cloud_access_token: str
-
         try:
-            config = GxAgentConfigSettings()
+            config = GxAgentServerSessionConfig()
+            self._agent_server_session_config = config
         except pydantic.ValidationError as validation_err:
             raise GXAgentError(
                 f"Missing or badly formed environment variable\n{validation_err.errors()}"
             ) from validation_err
 
         # obtain the broker url and queue name from Cloud
+        agent_server_session = AgentServerSession(config)
 
-        agent_sessions_url = f"{config.gx_cloud_base_url}/organizations/{config.gx_cloud_organization_id}/agent-sessions"
+        agent_sessions_url = agent_server_session.build_url(
+            resource_endpoint="/agent-sessions"
+        )
```

Comment:
thanks for the feedback @roblim, let's make this less confusing! the main drawback of the approach you propose is that we need to wrap all of the `Session` API - my goal was to make a pre-configured `Session` object available, and to have the code using the session (agent actions) and to be able to define general purpose config (like building the url) in a single place.

Reply:
specifically, would renaming the class to `CloudSession`, and renaming the `session` method to `create_session` make this abstraction easier to grok?

Reply:
I think that would be an improvement, maybe `GxCloudSession` to make things more clear? Def like `create_session` more. I also like @Kilo59's suggestion of `GxCloudClient`. 

Revisiting this again, I think more so than just the naming, it's all the disparate parts - I feel like there could be more encapsulation. Currently, in order to achieve the end of doing something with some GX Cloud endpoint using `AgentServerSession`, you need to: 
- Materialize a `GxAgentServerSessionConfig` object, then pass to `AgentServerSession` to instantiate an instance of `AgentServerSession`
- Use that `AgentServerSession` instance to materialize both a:
  - Session object and
  - Appropriate cloud endpoint
- Then call session.[___](url=url...)
- Handle/raise possible errors

I think `AgentServerSession` can at least handle materializing its own config if one isn't passed in. Given all of the above required steps to do X, it's not clear what we're getting by introducing a separate class, vs. say just having 2 functions called `get_agent_cloud_session` (which internally just figures out the right values, since `GxAgentServerSessionConfig` just seems to do just that - looks for env/global config vars) and `get_agent_cloud_url`, then using like above. Do you foresee `AgentServerSession` getting more functionality over time?

Reply:
Er, realizing the first point doesn't need to happen every time. Maybe I'm also confused because the currently available impl example is within `GXAgent` (`_update_status`, etc.), but intention is also to be called within `AgentAction` as well.